daemon/c_service.c

```diff
@@ -511,18 +527,58 @@ c_service_start_pre_exec(c_service_t *service)
  * Helper function that generates and sends a protobuf message to the Trustme Service.
  */
 static int
-c_service_send_message_proto(c_service_t *service, unsigned int code)
+c_service_send_message_proto(c_service_t *service, unsigned int code, const char *msg)
 {
 	ASSERT(service);
 
 	CmldToServiceMessage message_proto = CMLD_TO_SERVICE_MESSAGE__INIT;
 	message_proto.code = code;
 
+	if (msg) {
+		message_proto.msg = mem_strdup(msg);
+	}
+
+	int ret =
+		protobuf_send_message(service->sock_connected, (ProtobufCMessage *)&message_proto);
+
+	if (msg)
+		mem_free(message_proto.msg);
```

Comment:
see comment above, restore old msg pointer?!

Reply:
removed parameter as it is not needed anymore